### PR TITLE
C++ code to usaco-831.mdx

### PR DESCRIPTION
**Pull request by old-School18**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

**Review by envyaims — CHANGES_REQUESTED** [review]
make the function names more meaningful than "util" and use tabs. Also put comments for what each of the functions do

File: solutions/usaco-831.mdx [hunk 1]
```diff
@@ -10,6 +10,97 @@ author: Jesse Choe, Kevin Sheng, Benjamin Qi, Brad Ma, Qian Qian
...
+  return false;
+}
+int individualWins(string s[]) {
+  int wins = 0;
+  for (char ch = 'A'; ch <= 'Z'; ch++) {
+    wins += individualWinsUtil(ch, s);
+  }
+  return wins;
+}
+bool teamWinsUtil2(char ch1, char ch2, char x, char y, char z) {
```
Comment (envyaims) [inline solutions/usaco-831.mdx:42]: wots this

File: solutions/usaco-831.mdx [hunk 2]
```diff
@@ -10,6 +10,102 @@ author: Jesse Choe, Kevin Sheng, Benjamin Qi, Brad Ma, Qian Qian
...
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+//check if the cow ch won in any row or column or diagonal
+bool checkSingleCowWinner(char ch, string s[]) {
+  for (int i = 0; i < 3; i++) {
+    //check rows
+    if (s[i][0] == ch && s[i][0] == s[i][1] && s[i][0] == s[i][2]) 
+      return true;
```
Comment (Aadit-Ambadkar) [inline solutions/usaco-831.mdx:23]:
```suggestion
    if (s[i][0] == ch && s[i][0] == s[i][1] && s[i][0] == s[i][2]) {
      return true;
    }
```

File: solutions/usaco-831.mdx [hunk 3]
```diff
@@ -10,6 +10,102 @@ author: Jesse Choe, Kevin Sheng, Benjamin Qi, Brad Ma, Qian Qian
...
+    //check columns
+    if (s[0][i] == ch && s[0][i] == s[1][i] && s[0][i] == s[2][i]) 
+      return true;
```
Comment (Aadit-Ambadkar) [inline solutions/usaco-831.mdx:26]:
```suggestion
    if (s[0][i] == ch && s[0][i] == s[1][i] && s[0][i] == s[2][i]) {
      return true;
    }
```

File: solutions/usaco-831.mdx [hunk 4]
```diff
@@ -10,6 +10,102 @@ author: Jesse Choe, Kevin Sheng, Benjamin Qi, Brad Ma, Qian Qian
...
+  }
+  //check diagonals
+  if (s[0][0] == ch && s[0][0] == s[1][1] && s[0][0] == s[2][2]) 
+    return true;
+  if (s[0][2] == ch && s[0][2] == s[1][1] && s[0][2] == s[2][0]) 
+    return true;
```
Comment (Aadit-Ambadkar) [inline solutions/usaco-831.mdx:32]:
```suggestion
  if (s[0][0] == ch && s[0][0] == s[1][1] && s[0][0] == s[2][2]) {
    return true;
  }
  if (s[0][2] == ch && s[0][2] == s[1][1] && s[0][2] == s[2][0]) {
    return true;
  }
```

File: solutions/usaco-831.mdx [hunk 5]
```diff
@@ -10,6 +10,102 @@ author: Jesse Choe, Kevin Sheng, Benjamin Qi, Brad Ma, Qian Qian
...
+    if (x == ch1 && y == ch2) 
+      return true;
+    if (x == ch2 && y == ch1) 
+      return true;
+  } else if (y == z && x != y) {
+    if (y == ch1 && x == ch2) 
+      return true;
+    if (y == ch2 && x == ch1) 
+      return true;
+  }
```
Comment (Aadit-Ambadkar) [inline solutions/usaco-831.mdx:61]:
```suggestion
  if (x == y && x != z) {
    if (x == ch1 && z == ch2) {
      return true;
    }
    if (x == ch2 && z == ch1) {
      return true;
    }
  } else if (x == z && x != y) {
    if (x == ch1 && y == ch2) {
      return true;
    }
    if (x == ch2 && y == ch1) {
      return true;
    }
  } else if (y == z && x != y) {
    if (y == ch1 && x == ch2) {
      return true;
    }
    if (y == ch2 && x == ch1) {
      return true;
    }
  }
```

File: solutions/usaco-831.mdx [hunk 6]
```diff
@@ -10,6 +10,102 @@ author: Jesse Choe, Kevin Sheng, Benjamin Qi, Brad Ma, Qian Qian
...
+    if (checkIfWinners(ch1, ch2, s[0][i], s[1][i], s[2][i])) 
+      return true;
+  }
+  //check diagonals
+  if (checkIfWinners(ch1, ch2, s[0][0], s[1][1], s[2][2])) 
+    return true;
+  if (checkIfWinners(ch1, ch2, s[0][2], s[1][1], s[2][0])) 
+    return true;
+  return false;
+}
```
Comment (Aadit-Ambadkar) [inline solutions/usaco-831.mdx:81]: braces

File: solutions/usaco-831.mdx [hunk 7]
```diff
@@ -10,6 +10,102 @@ author: Jesse Choe, Kevin Sheng, Benjamin Qi, Brad Ma, Qian Qian
...
+  }
+  return wins;
+}
+int main() {
+  freopen("tttt.in", "r", stdin);
+  freopen("tttt.out", "w", stdout);
+  
+  string s[3];
+  for (int i = 0; i < 3; i++) 
+  	cin >> s[i];
```
Comment (Aadit-Ambadkar) [inline solutions/usaco-831.mdx:98]:
```suggestion
  for (int i = 0; i < 3; i++) {
  	cin >> s[i];
  }
```

File: solutions/usaco-831.mdx [hunk 8]
```diff
@@ -10,6 +10,102 @@ author: Jesse Choe, Kevin Sheng, Benjamin Qi, Brad Ma, Qian Qian
...
+
+  int individualCows = individualWins(s);
+  int teamCows = teamWins(s);
+
+  cout << individualCows << endl;
+  cout << teamCows << endl;
+}
```
Comment (Aadit-Ambadkar) [inline solutions/usaco-831.mdx:105]: tabs, not spaces! Fix the formatting!

**Review by Aadit-Ambadkar — CHANGES_REQUESTED** [review]
So much formatting needed

**Review by Aadit-Ambadkar — APPROVED** [review]

File: solutions/usaco-831.mdx [hunk 9]
```diff
@@ -10,6 +10,123 @@ author: Jesse Choe, Kevin Sheng, Benjamin Qi, Brad Ma, Qian Qian
 [Official Analysis (C++)](http://www.usaco.org/current/data/sol_tttt_bronze_open18.html)
 
 <LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+//check if the cow ch won in any row or column or diagonal
```
Comment (SansPapyrus683) [inline solutions/usaco-831.mdx:19]:
```suggestion
// check if the cow ch won in any row or column or diagonal
```
do the same for all the other comments

File: solutions/usaco-831.mdx [hunk 10]
```diff
@@ -10,6 +10,123 @@ author: Jesse Choe, Kevin Sheng, Benjamin Qi, Brad Ma, Qian Qian
...
+bool checkSingleCowWinner(char ch, string board[]) {
```
Comment (SansPapyrus683) [inline solutions/usaco-831.mdx:20]: i'd like it if you used vectors instead of arrays

File: solutions/usaco-831.mdx [hunk 11]
```diff
@@ -10,6 +10,123 @@ author: Jesse Choe, Kevin Sheng, Benjamin Qi, Brad Ma, Qian Qian
...
+	}
+	if (board[0][2] == ch && board[1][1] == ch && board[2][0] == ch) {
+		return true;
+	}
+	return false;
+}
+
+//determine the number of individual cow wins
+int individualWins(string board[]) {
+	int singleCowWins = 0;
```
Comment (SansPapyrus683) [inline solutions/usaco-831.mdx:43]: snake_case for c++ instead of camelCase

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Review by SansPapyrus683 — APPROVED** [review]
we gud

**Comment by old-School18** [comment]
> make the function names more meaningful than "util" and use tabs. Also put comments for what each of the functions do

the changes you suggested were completed. Anything else?

**Comment by Aadit-Ambadkar** [comment]
@SansPapyrus683 merge (2 accepted)

**Pull request merged**